src/App.tsx

```diff
@@ -41,55 +51,117 @@ export function getReorderedGoods(
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
+
+export class App extends React.Component<{}, State> {
+  state = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  alphabetSort = () => {
+    this.setState(() => ({
+      sortType: SortType.ALPHABET,
+    }));
+  };
+
+  lengthSort = () => {
+    this.setState(() => ({
+      sortType: SortType.LENGTH,
+    }));
+  };
```

Comment:
You don't need to use the function inside setState in those cases
```suggestion
    this.setState({ sortType: SortType.ALPHABET });
  };

  lengthSort = () => {
    this.setState({ sortType: SortType.LENGTH });
  };
```